plugins/text-editor-resources/src/components/extension/colors.ts

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright © 2025 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { getEventPositionElement, showPopup } from '@hcengineering/ui'
+import { type Editor } from '@tiptap/core'
+import ColorPicker from './popups/ColorPicker.svelte'
+import { type ActionContext } from '@hcengineering/text-editor'
+
+export interface BackgroundColorOptions {
+  types: string[]
+}
+
+interface ColorSpec {
+  color: string
+  preview?: string
+}
+
+function colorVar (tag: string, prefix = 'text'): string {
+  return `var(--theme-text-editor-palette-${prefix}-${tag})`
+}
+
+function colorSpec (tag: string, prefix = 'text'): ColorSpec {
+  const color = colorVar(tag, prefix)
+  return { color, preview: colorVar(tag) }
+}
+
+const palette = {
+  background: [
+    { color: 'transparent' },
+    colorSpec('gray', 'bg'),
```

Comment:
May be we could use our palette? colors.ts?

Reply:
There are a few problems with this. Both palettes are oriented to the same constraints (the same colors are used there for both themes), so black text will not be readable, besides that they are just too vibrant to be used in tables even for a black theme. We would also need a different rendering mechanism for these extensions to work with data represented in this way. Right now, the colors that are provided there are linked to CSS variable names, so we can update the palettes if needed in the future when they are adjusted for the needs of these contexts.